pkg/datasources/network_policies_acceptance_test.go

```diff
@@ -0,0 +1,200 @@
+package datasources_test
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"regexp"
+	"testing"
+
+	acc "github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/acceptance"
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/acceptance/helpers/random"
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/acceptance/testenvs"
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/provider/resources"
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/sdk"
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+	"github.com/hashicorp/terraform-plugin-testing/tfversion"
+	"github.com/stretchr/testify/require"
+)
+
+func TestAcc_NetworkPolicies_Complete(t *testing.T) {
+	_ = testenvs.GetOrSkipTest(t, testenvs.ConfigureClientOnce)
+
+	id := acc.TestClient().Ids.RandomAccountObjectIdentifier()
+	id2 := acc.TestClient().Ids.RandomAccountObjectIdentifier()
+	comment := random.Comment()
+
+	databaseId := acc.TestClient().Ids.RandomAccountObjectIdentifier()
+	schemaId := acc.TestClient().Ids.RandomDatabaseObjectIdentifierInDatabase(databaseId)
+
+	allowedNetworkRuleId1 := acc.TestClient().Ids.RandomSchemaObjectIdentifierInSchema(schemaId)
+	allowedNetworkRuleId2 := acc.TestClient().Ids.RandomSchemaObjectIdentifierInSchema(schemaId)
+	blockedNetworkRuleId1 := acc.TestClient().Ids.RandomSchemaObjectIdentifierInSchema(schemaId)
+	blockedNetworkRuleId2 := acc.TestClient().Ids.RandomSchemaObjectIdentifierInSchema(schemaId)
+
+	resource.Test(t, resource.TestCase{
+		ProtoV6ProviderFactories: acc.TestAccProtoV6ProviderFactories,
+		TerraformVersionChecks: []tfversion.TerraformVersionCheck{
+			tfversion.RequireAbove(tfversion.Version1_5_0),
+		},
+		CheckDestroy: acc.CheckDestroy(t, resources.NetworkPolicy),
+		Steps: []resource.TestStep{
+			{
+				PreConfig: func() {
+					db, _ := acc.TestClient().Database.CreateDatabaseWithIdentifier(t, databaseId)
```

Comment:
Why can't test database be reused here?

Reply:
It wasn't created but that's my bad because the test didn't have a precheck (because of the precheck we have dependency to the package where the common database,schema,etc. are created).